test/test_distributions.py

```diff
@@ -599,6 +599,13 @@ def test_invalid_parameter_broadcasting(self):
         for dist, kwargs in invalid_examples:
             self.assertRaises(RuntimeError, dist, **kwargs)
 
+    def test_rsample_backward_smoke(self):
+        for Dist, params in EXAMPLES:
+            for i, param in enumerate(params):
+                if Dist.has_rsample and all(isinstance(p, Variable) for p in param):
+                    dist = Dist(**param)
+                    dist.rsample().sum().backward()
```

Comment:
Can you check that params got any gradients? Also, it's nicer to use `torch.autograd.grad`, because otherwise this test will have a side effect!